### PR TITLE
migrate logstash-core-event-java to logstash-core

### PR DESCRIPTION
**Pull request by talevy**
This PR deletes the `logstash-core-event-java` project/gem and migrates all its code into `logstash-core`.

### CODE REVIEW

File: logstash-core/lib/logstash-core/logstash-core.rb [hunk 1]
```diff
@@ -23,3 +23,8 @@ module LogStash
     raise("Error loading logstash-core/logstash-core.jar file, cause: #{e.message}")
   end
 end
+
+require "jruby_event_ext"
+require "jruby_timestamp_ext"
+require "logstash/event"
```
Comment (talevy) [inline logstash-core/lib/logstash-core/logstash-core.rb]: might make sense to move these, but not sure where. environment?

File: logstash-core/src/test/java/org/logstash/EventTest.java [hunk 2]
```diff
@@ -1,5 +1,3 @@
-package org.logstash;
```
Comment (talevy) [inline logstash-core/src/test/java/org/logstash/EventTest.java]: woops. needs fixing. wrong package

**Comment by colinsurprenant** [comment]
Good job! Overall changes LGTM - did not test yet.

**Comment by talevy** [comment]
@colinsurprenant plan on testing(so that is not an LGTM)?

**Comment by colinsurprenant** [comment]
yes I was planning on testing,  but if all builds + tests + packaging have been successfully tested on your side I am good.

**Comment by talevy** [comment]
@colinsurprenant isn't that what our tests are for? we package it up for running RATS tests.

**Comment by talevy** [comment]
Merged into `5.x` ^^ with this commit: https://github.com/elastic/logstash/commit/7c75b598143fa24490bc4e54bdb8501032642dc9

**Comment by talevy** [comment]
follow-up PRs: 

- remove legacy logstash-core https://github.com/elastic/logstash/pull/6783
- remove core-queue https://github.com/elastic/logstash/pull/6782

**Pull request merged**